src/models/components/policy_value_common_net/policy_value_common_net.py

```diff
@@ -0,0 +1,31 @@
+import torch
+import torch.distributions as distributions
+import torch.nn as nn
+from torch import Tensor
+
+from ..stochastic_policy.stochastic_policy import StochasticPolicy
+from ..value.value import Value
+
+
+class PolicyValueCommonNet(nn.Module):
+    def __init__(self, policy: StochasticPolicy, value: Value, height: int, width: int, dim_hidden: int):
+        super().__init__()
+        self.policy = policy
+        self.value = value
+        self.conv2d1 = nn.Conv2d(3, 32, 8, stride=4)
+        self.conv2d2 = nn.Conv2d(32, 64, 4, stride=2)
+        self.conv2d3 = nn.Conv2d(64, 64, 3, stride=1)
+        self.fc = nn.Linear(
+            ((((height - (8 - 4)) // 4 - (4 - 2)) // 2 - (3 - 1)) // 1)
+            * ((((width - (8 - 4)) // 4 - (4 - 2)) // 2 - (3 - 1)) // 1)
+            * 64,
+            dim_hidden,
+        )
```

Comment:
Dependency Injectionかけてください。
```suggestion
    def __init__(self, base_model: nn.Module, policy: StochasticPolicy, value: Value, height: int, width: int, dim_hidden: int):
        super().__init__()
        self.policy = policy
        self.value = value
        self.base_model = base_model
```

Reply:
base_modelが他クラスと共有されるようなことがないのであればPolicyValueCommonNetを抽象クラスとしてConv2d等実装は継承したクラスでやってもよいかとも思いましたがどうでしょうか

Reply:
@myxyy 継承先でやる提案はOKです。そうしてください。
ただ、モデルの構造を後から変化させたい場合に現在の直書きするやり方は好まれません。また、InverseDynamicsを作る過程で、`CNNObservationEncoder`を実装しており、内部構造がほとんどそれと一致しています。それを再利用可能な構造にするべきだと思います。

Reply:
Conv部分を再利用するようならばDIの方が良いですね
SmallConvNetをutilsに作りDIするようにしました  
https://github.com/MLShukai/PrimitiveAMI/pull/16/commits/7f7abed66b6ce4de97d9c4931ddb5e111760aa4e  
https://github.com/MLShukai/PrimitiveAMI/pull/16/commits/605936b22a3a5c48eb5829e322e6c27d8c965d4f